blog: centralized logging on ceph dashboard (#472)
ceph/ceph.io: src/en/news/blog/2022/centralized_logging/index.md

---
title: "Centralized Logging on Ceph Dashboard"
date: "2022-11-23"
author: "Nizamudeen A"
image: "images/banner.jpg"
tags:
  - "ceph"
  - "dashboard"
  - "logs"
  - "monitoring"
  - "loki"
  - "promtail"
  - "grafana"
  - "centralized logging"
---

_Disclaimer: This article proclaims Loki as a hero._

## What are we trying to solve?

> Ceph, as a distributed system, usually generates a lot of logs. And these logs can be from mons, mgr, osds or any of the clients of ceph.

Reviewer: add a comma after osds.

Reviewer 2: `OSDs` since it's an acronym, and capitalize Ceph.

Naturally, we look at the logs to get some detailed information regarding the events that happened throughout the cluster lifecycle.

> Logs are what we immediately look for if we find something unusual going on in the cluster. Or if you like reading, its a fun place to be 😜.

Reviewer: s/its/it's/

Reviewer 2: OSDs , Ceph

> In anycase, we'll want to check through logs from different clients and they are not available from a single source. Its all distributed as it is intended.

Reviewer: ```suggestion
In any case, we'll want to check through logs from different clients, which are unavailable from a single source. It's all distributed as intended.
```

Reviewer 2: I agree with Thingee's suggestion, other than that I would leave "not available"

> In anycase, we'll want to check through logs from different clients and they are not available from a single source. Its all distributed as it is intended.

Author: Done, thanks

## Enter Loki.

> Thankfully, the Loki we are dealing with is not an Asgardian God(imagine the chaos). Our Loki is a log aggregation system which is designed to store and

Reviewer 3: /s/which/that/

or

s/system which is designed to store/system designed to store/

> query the logs. Infact it is not just Loki, it is [Grafana Loki](https://grafana.com/oss/loki/).

Reviewer 4: ```suggestion
query the logs. In fact, it is not just Loki, it is [Grafana Loki](https://grafana.com/oss/loki/).
```

> query the logs. Infact it is not just Loki, it is [Grafana Loki](https://grafana.com/oss/loki/).

Reviewer: ```suggestion
query the logs. In fact, it is not just Loki; it is [Grafana Loki](https://grafana.com/oss/loki/).
```

We have a Promtail too.

![tony_stark_promtail](images/we_have_promtail.png)

> [Promtail](https://grafana.com/docs/loki/latest/clients/promtail/) is an agent which ships the contents of local logs to a private Grafana Loki instance.

Reviewer: ```suggestion
[Promtail](https://grafana.com/docs/loki/latest/clients/promtail/) is an agent that ships the contents of local logs to a private Grafana Loki instance.
```

> Let's make it simple enough. We introduced a feature called Centralized Logging, where we can go to a particular section in the Ceph Dashboard and see the

Reviewer 3: s/simple enough/simple/

logs from all the clients in just one place. Cool, isn't it?

## How to set it up?

> First, we have to make sure the grafana is configured on our cluster.

Reviewer 3: s/to make sure the Grafana/to make sure that Grafana/

Then, go to the **Cluster > Services** page, and **Create** the two services - Loki and Promtail.

![services_list](images/loki_promtail.png)

If successful, we can see all the daemons are **running**.

##### Status of Loki Daemon

![loki_daemon](images/loki.png)

##### Status of Promtail Daemons

![promtail_daemon](images/promtail.png)

Now we need to enable logging to the files. For that, we can go to **Cluster > Configurations**,
search for the **log_to_file** and **mon_cluster_log_to_file**, in the search area

![log_to_file](images/log_to_file.png)

Just select and **Edit** the option. And select the value of **global** to true

![config_form](images/form_config.png)

We have to do the same for **mon_cluster_log_to_file** as well.

![mon_cluster_log_to_file](images/mon_log_to_file.png)

> If you can't find the item you searched for, then you can just **clear the filter** and it will bring you the configuration option that you searched.

Reviewer: ```suggestion
If you can't find the item you searched for, then you can just **clear the filter** and it will bring you the configuration option that you searched for.
```

![filter](images/filter.png)

### Viewing the Logs

> Go to **Cluster > Logs** and go to the **Daemon Logs** tab. That is where we are going to see the Centralized logs.

Reviewer 3: s/where we are going to see/where we find/

> If you see a screen like the below image, then you have to sign in to the grafana and just reload the page again

Reviewer 3: s/to the grafana/to Grafana/

> If you see a screen like the below image, then you have to sign in to the grafana and just reload the page again

Reviewer: ```suggestion
If you see a screen like the below image, then you have to sign in to the grafana and just reload the page again.
```

![not_signed_in_view](images/error_page.png)

> If you are signed in, then we can see the logs view directly.

Reviewer 3: s/we can see/you can see/
